pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -192,6 +192,11 @@ func (spc *realStatefulPodControl) createPersistentVolumeClaims(set *apps.Statef
 		case err != nil:
 			errs = append(errs, fmt.Errorf("Failed to retrieve PVC %s: %s", claim.Name, err))
 			spc.recordClaimEvent("create", set, pod, &claim, err)
+		case err == nil:
```

Comment:
I think there is a race condition with this fix as @msau42 already [said](https://github.com/kubernetes/kubernetes/issues/74374#issue-413151492), if the pvc can have DeletionTimeStamp after this method is run, and then pod will be pending state again?

Reply:
Since this is not an atomic operation, I think creating statefulSet pod can not avoid the scene absolutely.  However, it seldom(almost never) would happen.